compiler/p/codegen/PPCSystemLinkage.cpp

```diff
@@ -625,6 +625,13 @@ TR::PPCSystemLinkage::mapSingleAutomatic(
 void
 TR::PPCSystemLinkage::createPrologue(TR::Instruction *cursor)
    {
+   TR::Node *firstNode = comp()->getStartTree()->getNode();
+
+   if (comp()->getOption(TR_EntryBreakPoints))
+      {
+      cursor = generateInstruction(cg(), TR::InstOpCode::bad, firstNode, cursor);
```

Comment:
According to our friends at the V8 project the breakpoint instruction gdb generates is `0x7D821008`. Will `TR::InstOpCode::bad` generate the same binary encoding? If not we need a proper pseudo instruction for this. See #2666 for how this was done on Z. It should be similar on Power.

[1] https://github.com/v8/v8/blob/a062708467a313d64818b5f8a0ac37dae8775959/src/ppc/assembler-ppc.cc#L1637

Reply:
Also I'm not sure if we want to generate something different for AIX in case people use DBX. I'll let @gita-omr answer that one as she knows best what people typically use.

Reply:
> According to our friends at the V8 project the breakpoint instruction gdb generates is 0x7D821008.

This is a trap instruction. We've traditionally not used them for breakpoints because traps are used to implement NULLCHKs, BOUNDCHKs, and so on in OpenJ9. `bad` just maps to an illegal instruction, which works just as well as a trap to signal the debugger, but makes continuing the program a little more awkward.

You can get away with using a trap in this case currently, because this code wont be used in OpenJ9, but from a more general perspective traps are useful for more than just debugger signalling on Power. As far as the OMR compiler goes I'd give some thought to whether or not you want to take over traps for debugging purposes, leave them for downstream, have it be configurable, etc.

Re: AIX, it works the same as GDB in this context.

Reply:
Thanks @ymanton for the background.